### DISCUSSION TITLE
[v13.1.4] Middleware not working on vercel

### DISCUSSION THREAD

**Discussion by Alek07** [vercel/next.js#45136] [Help]
I created a middleware to protect my routes whenever a user is logged in or not and it looks like this
```
import { getToken } from "next-auth/jwt"
import { NextRequest, NextResponse } from "next/server"

export const middleware = async (req: NextRequest) => {
    const token = await getToken({ req, secret: process.env.NEXT_PUBLIC_SECRET })
    const { pathname } = req.nextUrl

    if (token && pathname === "/") {
        return NextResponse.redirect(new URL('/dashboard', req.url))
    }

    if (pathname.includes("/api/auth") || token) {
        return NextResponse.next()
    }

    if (!token && pathname !== "/") {
        return NextResponse.redirect(new URL('/', req.url))
    }
}

export const config = {
    matcher: ["/dashboard", "/artists", "/tracks", "/"],
}
```
it works on my local environment but when I deploy it on vercel the middleware does not work as expected. Also, I have the `middleware.ts` in the root level of the project as suggested but still not working.

**Comment by ijjk** [comment]
Hi, following up here, were you able to track down the issue with this? We've seen this occur from `next export` being used or a misconfigured project setting which should now have warnings in the latest version of Next.js.

**Reply by Alek07** [reply]
Hey, I haven't been lucky with this one, I'm using Next 13.1.6 and still have the issue on my vercel deploy environment.

**Reply by ijjk** [reply]
I'd be happy to take a look, can you share a deployment/repo?

**Reply by Alek07** [reply]
Sure, this is the repo I'm working on https://github.com/Alek07/myspotify-tracker-client

**Reply by ijjk** [reply]
It looks like the middleware is deployed correctly on that repo, can you share additional context on what is not behaving the same as local?

**Reply by Alek07** [reply]
Oh sorry, I have a patch to make it work on the project, can you try again from the development branch? The expected behavior is when you click on a link from the top menu and the user is not authenticated, it should not let the user access any other route.

**Reply by ijjk** [reply]
Is this only occurring in the latest version of Next.js also does the client transition issue occur without middleware as well?

**Reply by Alek07** [reply]
I'd been having this issue on 13.0.1, 13.1.4, and now 13.1.6. Also, without the middleware it let the user navigate to the other routes, which is why the middleware was set up to prevent that, so being able to navigate with the middleware is not the expected result.

**Comment by Saszr** [comment]
Hmmm, feels like the problem I have encountered. What progress has been made now?

This is a discussion I started in the Vercel: https://github.com/orgs/vercel/discussions/2168
